src/core/service.c

```diff
@@ -577,8 +577,8 @@ static int service_verify(Service *s) {
                 return -ENOEXEC;
         }
 
-        if (s->type == SERVICE_ONESHOT && s->restart != SERVICE_RESTART_NO) {
-                log_unit_error(UNIT(s), "Service has Restart= setting other than no, which isn't allowed for Type=oneshot services. Refusing.");
+        if (s->type == SERVICE_ONESHOT && !IN_SET(s->restart, SERVICE_RESTART_NO, SERVICE_RESTART_ON_FAILURE, SERVICE_RESTART_ON_ABNORMAL, SERVICE_RESTART_ON_WATCHDOG, SERVICE_RESTART_ON_ABORT)) {
```

Comment:
Hmm, maybe move the IN_SET() check to a new line? It is pretty hard to read now.
Also, maybe it'd be easier to invert the condition, so that it matches the error message?
```C
if (s->type == SERVICE_ONETSHOT &&
    IN_SET(s->restart, SERVICE_RESTART_ALWAYS, SERVICE_RESTART_ON_SUCCESS))
   ...
```

Reply:
I think it's best to have it explicit rather than implicit so we do not accidentally allow new conditions. But if the general consensus is to go the other way I'll change it.

Reply:
It's always a bit risky to have an error message that does not match the condition being tested... But here there's no clear answer, do whatever you think is better.

Reply:
Keeping the condition the way it is but updating the error message to match (even though it would more verbose) seems to better match the surrounding cases.